pac4j-http/src/main/java/org/pac4j/http/credentials/authenticator/RestAuthenticator.java

```diff
@@ -0,0 +1,142 @@
+package org.pac4j.http.credentials.authenticator;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.pac4j.core.context.HttpConstants;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.credentials.UsernamePasswordCredentials;
+import org.pac4j.core.credentials.authenticator.Authenticator;
+import org.pac4j.core.exception.CredentialsException;
+import org.pac4j.core.exception.HttpAction;
+import org.pac4j.core.exception.TechnicalException;
+import org.pac4j.core.profile.definition.CommonProfileDefinition;
+import org.pac4j.core.profile.definition.ProfileDefinitionAware;
+import org.pac4j.core.util.CommonHelper;
+import org.pac4j.core.util.HttpUtils;
+import org.pac4j.http.profile.RestProfile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Authenticates against a REST API. The username/password are passed as a basic auth via a POST request,
+ * the JSON response is a user profile.
+ *
+ * @author Jerome Leleu
+ * @since 2.1.0
+ */
+public class RestAuthenticator extends ProfileDefinitionAware<RestProfile> implements Authenticator<UsernamePasswordCredentials> {
+
+    private static final Logger logger = LoggerFactory.getLogger(RestAuthenticator.class);
+
+    private static ObjectMapper mapper;
+
+    static {
+        mapper = new ObjectMapper();
```

Comment:
this should be injectable I think (with a default lazy-constructed static instance in case none is provided), because many applications have their own objectmapper configured...